File: Data Structure/String/Medium/LET_no522_Longest Uncommon Subsequence II.cpp
class Solution {
public:
    int findLUSlength(vector<string>& strs) {
        sort(strs.begin(), strs.end(), [](string& a, string& b) { return a.size() > b.size(); });
        for (string& sub : strs) {
            int count = 0;
            for (string& target : strs)
                count += issubsequence(sub, target);
            if (count == 1)
                return sub.size();
        }
        return -1;
    }
    
    int issubsequence(string& sub, string& target) {
        int i = 0;
        for (char c : target) {
            if (c == sub[i])
                i++;
        }
        return i == sub.size();
    }
};

File: Data Structure/String/Medium/LET_no522_Longest Uncommon Subsequence II.py
class Solution:
    def findLUSlength(self, strs: List[str]) -> int:
        def issubsequence(sub, target):
            it  = iter(target)
            return all(c in it for c in sub)
        
        for sub in sorted(strs, key=len, reverse=True):
            if sum(issubsequence(sub, target) for target in strs) == 1:
                return len(sub)
        
        return -1